test/test_store/test_store_berkeleydb.py

```diff
@@ -1,130 +1,147 @@
+from rdflib.store import VALID_STORE
 
-
-class BerkeleyDBTestCase(unittest.TestCase):
-    def setUp(self):
-        if not has_bsddb:
-            self.skipTest("skipping as berkleydb is missing")
-        self.store_name = "BerkeleyDB"
-        self.path = mktemp()
-        self.g = ConjunctiveGraph(store=self.store_name)
-        self.rt = self.g.open(self.path, create=True)
-        assert self.rt == VALID_STORE, "The underlying store is corrupt"
-        assert (
-            len(self.g) == 0
-        ), "There must be zero triples in the graph just after store (file) creation"
-        data = """
-                PREFIX : <https://example.org/>
-
-                :a :b :c .
-                :d :e :f .
-                :d :g :h .
-                """
-        self.g.parse(data=data, format="ttl")
-
-    def tearDown(self):
-        self.g.close()
-
-    def test_write(self):
-        assert (
-            len(self.g) == 3
-        ), "There must be three triples in the graph after the first data chunk parse"
-        data2 = """
-                PREFIX : <https://example.org/>
-
-                :d :i :j .
-                """
-        self.g.parse(data=data2, format="ttl")
-        assert (
-            len(self.g) == 4
-        ), "There must be four triples in the graph after the second data chunk parse"
-        data3 = """
-                PREFIX : <https://example.org/>
-
-                :d :i :j .
-                """
-        self.g.parse(data=data3, format="ttl")
-        assert (
-            len(self.g) == 4
-        ), "There must still be four triples in the graph after the thrd data chunk parse"
-
-    def test_read(self):
-        sx = None
-        for s in self.g.subjects(
-            predicate=URIRef("https://example.org/e"),
-            object=URIRef("https://example.org/f"),
-        ):
-            sx = s
-        assert sx == URIRef("https://example.org/d")
-
-    def test_sparql_query(self):
-        q = """
+pytestmark = pytest.mark.skipif(
+    not has_bsddb, reason="skipping berkeleydb tests, modile not available"
+)
+
+
+@pytest.fixture
+def get_graph():
+    path = mktemp()
+    g = ConjunctiveGraph("BerkeleyDB")
+    rt = g.open(path, create=True)
+    assert rt == VALID_STORE, "The underlying store is corrupt"
+    assert (
+        len(g) == 0
+    ), "There must be zero triples in the graph just after store (file) creation"
+    data = """
             PREFIX : <https://example.org/>
 
-            SELECT (COUNT(*) AS ?c)
-            WHERE {
-                :d ?p ?o .
-            }"""
+            :a :b :c .
+            :d :e :f .
+            :d :g :h .
+            """
+    g.parse(data=data, format="ttl")
 
-        c = 0
-        for row in self.g.query(q):
-            c = int(row.c)
-        assert c == 2, "SPARQL COUNT must return 2"
+    yield path, g
 
-    def test_sparql_insert(self):
-        q = """
-            PREFIX : <https://example.org/>
+    g.close()
+    g.destroy(path)
```

Comment:
Destroy was not in the old test, also it takes a configuration object and not path, and it won't do anything in this case.

I guess it does not hurt, but it also is not clear that this is the required usage.

Reply:
Uh-huh, okay. I hadn't noticed that the Store spec had got amended along the way somewhere. Originally, `destroy` had `configuration` as a positional param: https://github.com/RDFLib/rdflib/blob/5cfd4c824206667de3c38b97e5db566a32c1a377/rdflib/store.py#L188.